README.md

```diff
@@ -2658,6 +2556,114 @@ end
 Note: If you are using Sinatra in the modular style, replace
 `Sinatra::Application` above with the class name of your app.
 
+## Security
+
+#### Session Secret Security
+
+To improve security, the session data in the cookie is signed with a session
+secret using `HMAC-SHA1`. This session secret should optimally be a
+cryptographically secure random value of an appropriate length which for
+`HMAC-SHA1` is greater than or equal to 64 bytes (512 bits, 128 hex
+characters). You would be advised not to use a secret that is less than 32
+bytes of randomness (256 bits, 64 hex characters). It is therefore **very
+important** that you don't just make the secret up, but instead use a secure
+random number generator to create it. Humans are extremely bad at generating
+random values.
+
+By default, a 32 byte secure random session secret is generated for you by
+Sinatra, but it will change with every restart of your application. If you
+have multiple instances of your application, and you let Sinatra generate the
+key, each instance would then have a different session key which is probably
+not what you want.
+
+For better security and usability it's
+[recommended](https://12factor.net/config) that you generate a secure random
+secret and store it in an environment variable on each host running your
+application so that all of your application instances will share the same
+secret. You should periodically rotate this session secret to a new value.
+Here are some examples of how you might create a 64-byte secret and set it:
+
+**Session Secret Generation**
+
+```text
+$ ruby -e "require 'securerandom'; puts SecureRandom.hex(64)"
+99ae8af...snip...ec0f262ac
+```
+
+**Session Secret Generation (Bonus Points)**
+
+Use the [sysrandom gem](https://github.com/cryptosphere/sysrandom#readme) to
```

Comment:
Some users may be put off by the fact that that GitHub repository is _archived_. I was not able to locate a successor repository. None of the [forks](https://github.com/cryptosphere/sysrandom/network/members) seem to have taken over.

Reply:
It looks like no successor is needed. I think references to the sysrandom gem should be removed from the README.

From https://github.com/cryptosphere/sysrandom#why

>System/OS-level random number generators like `/dev/urandom` and `getrandom(2)` provide the best option for generating cryptographically secure random numbers.
>
>Ruby's built-in SecureRandom does not provide this, but instead uses OpenSSL's userspace RNG. This has been a [source of vulnerabilities](https://emboss.github.io/blog/2013/08/21/openssl-prng-is-not-really-fork-safe/) in Ruby, and this [Ruby bug ticket](https://bugs.ruby-lang.org/issues/9569) contains more discussion on the issue.

The Ruby bug ticket, https://bugs.ruby-lang.org/issues/9569, has been closed since 5 years. It was closed with this commit, available from Ruby 2.5: https://github.com/ruby/ruby/commit/abae70d6ed63054d7d01bd6cd80c1b5b98b93ba3

Reply:
@dentarg Thanks for the deep dive!